### PR TITLE
Add Notification Service - closes #1860

### LINKED ISSUE #1860: Add Notification Service
Part of the Notification task

size Medium

### PR DESCRIPTION
**Pull request by danieledamiani**

### CODE REVIEW

**Comment by iamvukasin** [comment]
Reviewing

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts [hunk 1]
```diff
@@ -0,0 +1,308 @@
+import { act } from 'react-dom/test-utils';
```
Comment (lazarmitic) [inline packages/tempus-client_v3/src/services/NotificationService.spec.ts:1]: I think `act` should be imported from `@testing-library/react`.

**Review by lazarmitic — APPROVED** [review]

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+  {
+    category: 'Transaction',
+    chain: 'ethereum',
+    content: 'content 3',
+    dismissed: false,
+    id: 'stored003',
+    level: 'warning',
+    timestamp: 1652569200010,
+    title: 'StoredNotification 3',
+  },
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/services/NotificationService.spec.ts:43]: I would extract these notifications into separate constants and use them where needed. E.g.
```
const notification1 = { ... };
const notification2 = { ... };
const notification3 = { ... };

const mockStoredNotifications = [notification1, notification2, notification3];
```

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+      expect(mockListener).toHaveBeenNthCalledWith(1, {
+        category: 'Transaction',
+        chain: 'ethereum',
+        content: 'content 3',
+        dismissed: false,
+        id: 'stored003',
+        level: 'warning',
+        timestamp: 1652569200010,
+        title: 'StoredNotification 3',
+      });
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/services/NotificationService.spec.ts:222]: (to continue the previous comment)

... and use the constants here:
```suggestion
      expect(mockListener).toHaveBeenNthCalledWith(1, notification3);
```

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+        {
+          category: 'Transaction',
+          chain: 'ethereum',
+          content: 'content 2',
+          dismissed: true, // this changed
+          id: 'stored002',
+          level: 'warning',
+          timestamp: 1652569200001,
+          title: 'StoredNotification 2',
+        },
```
Comment (iamvukasin) [inline packages/tempus-client_v3/src/services/NotificationService.spec.ts:294]: Even in this case it is as simple as 
```suggestion
        {
          ...notification2,
          dismissed: true, // this changed
        },
```

**Review by iamvukasin — DISMISSED** [review]
Optional test readability improvements. Otherwise, looks good

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts (hunk 1 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/services/NotificationService.spec.ts:1]: in other tests is imported from `react-dom` but anywat it does the same thing

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts (hunk 4 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/services/NotificationService.spec.ts]: refactored

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts (hunk 3 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/services/NotificationService.spec.ts]: refactored

File: packages/tempus-client_v3/src/services/NotificationService.spec.ts (hunk 2 above)
Comment (danieledamiani) [reply packages/tempus-client_v3/src/services/NotificationService.spec.ts]: refactored

**Review by iamvukasin — APPROVED** [review]

**Pull request merged**